src/braket/default_simulator/simulator.py

```diff
@@ -407,7 +442,12 @@ def _formatted_measurements(
         ]
         #  Gets the subset of measurements from the full measurements
         if measured_qubits is not None and measured_qubits != []:
-            if any(qubit in range(simulation.qubit_count) for qubit in measured_qubits):
+            if max(measured_qubits) != len(measured_qubits) - 1:
```

Comment:
This condition to identify that the measured qubits are not contiguous relies on the assumption that `measured_qubits` is sorted. I'd suggest removing this condition. Really, we don't make a mistake by applying the qubit_map in all cases; there no need for this condition.